src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +32,15 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
+        //Preparation and conversion from string to array of string 
+        //Assignment of array element to Class elements
+        address = address.replaceAll(", ", " ");
+        String[] splitArray = address.split("\\s+");
+        
+        block = new Block(splitArray[0]);
```

Comment:
Don't see any documentation saying you are enforcing 4 components. Thus this will throw an ArrayIndexOutOfBound exception. Handle it.